### PR TITLE
[PRM-1482] Updated Penalties Connector with new parameter

### PR DESCRIPTION
**Pull request by JBlacklaws**

### CODE REVIEW

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/config/featureSwitches/FeatureSwitch.scala [hunk 1]
```diff
@@ -22,7 +22,7 @@ sealed trait FeatureSwitch {
 
 object FeatureSwitch {
   val prefix: String = "feature.switch"
-  val listOfAllFeatureSwitches: List[FeatureSwitch] = List(CallAPI1812ETMP, UseAPI1812Model)
+  val listOfAllFeatureSwitches: List[FeatureSwitch] = List(CallAPI1812ETMP, UseAPI1812Model, UseAPI1811Model)
```
[the remark this answers is not in the document]
Comment (JBlacklaws) [reply app/config/featureSwitches/FeatureSwitch.scala:25]: I was thinking we might need this feature switch in the future when implementing the new parameters in the views, but I guess that's jumping the gun

File: app/connectors/PenaltiesConnector.scala [hunk 2]
```diff
@@ -29,16 +29,23 @@ class PenaltiesConnector @Inject()(httpClient: HttpClient,
...
-    httpClient.GET[ETMPPayload](s"$penaltiesBaseUrl${getPenaltiesDataUrl(enrolmentKey, isUsingNewApi)}")
+  def getPenaltiesData(enrolmentKey: String, isUsingNewApi: Boolean = false,
+                       isUsingNewFinancialApiModel: Boolean = false)(implicit user: User[_], hc: HeaderCarrier): Future[ETMPPayload] = {
+    httpClient.GET[ETMPPayload](s"$penaltiesBaseUrl${getPenaltiesDataUrl(enrolmentKey, isUsingNewApi, isUsingNewFinancialApiModel)}")
   }
 
-  def getPenaltyDetails(enrolmentKey: String, isUsingNewApi: Boolean = true)(implicit user: User[_], hc: HeaderCarrier): Future[GetPenaltyDetailsResponse] = {
-    httpClient.GET[GetPenaltyDetailsResponse](s"$penaltiesBaseUrl${getPenaltiesDataUrl(enrolmentKey, isUsingNewApi)}")(GetPenaltyDetailsResponseReads, hc, ec)
+  def getPenaltyDetails(enrolmentKey: String, isUsingNewApi: Boolean = true,
+                        isUsingNewFinancialApiModel: Boolean = true)(implicit user: User[_], hc: HeaderCarrier): Future[GetPenaltyDetailsResponse] = {
```
[the remark this answers is not in the document]
Comment (JBlacklaws) [reply app/connectors/PenaltiesConnector.scala:47]: And if it's not hard coded, wouldn't this be where we can use the feature switch?

**Comment by LewisMitchell** [comment]
Also could you resolve the conflicts, please?

File: app/connectors/PenaltiesConnector.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (JBlacklaws) [reply app/connectors/PenaltiesConnector.scala:47]: Righto, I'll implement the feature switch on the service to determine if the boolean is true or false

File: it/connectors/PenaltiesConnectorISpec.scala [hunk 3]
```diff
@@ -17,22 +17,32 @@
...
 import org.scalatest.concurrent.ScalaFutures.convertScalaFuture
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
+import stubs.PenaltiesStub
 import stubs.PenaltiesStub._
 import testUtils.IntegrationSpecCommonBase
 import uk.gov.hmrc.http.HeaderCarrier
 
-class PenaltiesConnectorISpec extends IntegrationSpecCommonBase {
+class PenaltiesConnectorISpec extends IntegrationSpecCommonBase with FeatureSwitching {
```
Comment (LewisMitchell) [inline it/connectors/PenaltiesConnectorISpec.scala:33]: I think you can remove the FeatureSwitching trait here as you bring it in transitively from IntegrationSpecCommonBase
```suggestion
class PenaltiesConnectorISpec extends IntegrationSpecCommonBase {
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Review by LewisMitchell — APPROVED** [review]
LGTM, just waiting on PRM-1480

**Pull request merged**